### PR TITLE
Adjust AMQP spec to changes in #521

### LINKED ISSUE #519: Relation between AMQP Format and Transport Binding is unclear
I've been reviewing the AMQP related parts of the spec, and I'm having trouble figuring out how the [data section](https://github.com/cloudevents/spec/blob/master/amqp-format.md#3-data) of the AMQP format relates to the AMQP transport binding. I found [this](https://github.com/cloudevents/spec/blob/master/amqp-transport-binding.md#14-event-formats):

> All implementations of this specification MUST support the JSON event format as well as the AMQP event format for the application-properties section

This restricts the usage of the AMQP event format to the application-properties section, which excludes the handling of `data`.

In [section 2.2](https://github.com/cloudevents/spec/blob/master/amqp-transport-binding.md#22-data) I found:
>An application is free to hold the information in any in-memory representation of its choosing, but as the value is transposed into AMQP as defined in this specification, the assumption is that the data value is made available as a sequence of bytes.

Finally [section 3.1.2](https://github.com/cloudevents/spec/blob/master/amqp-transport-binding.md#312-event-data-encoding) simply states:
>The data byte-sequence is used as the AMQP application-data section.

So the handling of `data` as described in the AMQP format with its usage of AMQP values would not take place. As I don't think the AMQP format would make sense for the structured encoding, either, this pretty much excludes that this rule from the AMQP format is applied:
>For other types (non-binary data without a datacontenttype attribute), the implementation MUST translate the data value into an AMQP type system value and the value MUST be stored in an AMQP value section.

All this really confuses me. In the examples section of the JSON format there is this example:

```
{
    "specversion" : "1.0-rc1",
    "type" : "com.example.someevent",
    "source" : "/mycontext",
    "id" : "C234-1234-1234",
    "time" : "2018-04-05T17:31:00Z",
    "comexampleextension1" : "value",
    "comexampleothervalue" : 5,
    "datacontenttype" : "application/json",
    "data" : {
        "appinfoA" : "abc",
        "appinfoB" : 123,
        "appinfoC" : true
    }
}
```
How would an SDK transform this into AMQP format/binary encoding? What would happen, if the `datacontenttype` was omitted? I suppose, according to the AMQP format some kind of AMQP value would be created. Would it be an AMQP Map?

### PR DESCRIPTION
**Pull request by deissnerk**
This PR assumes the changes of #521 to be accepted and fixes #519. It renames the AMQP transport binding into protocol mapping.
With the clear definition of event format in the main spec, a separate AMQP event format became obsolete. I copied the type mapping for the attributes over to the AMQP protocol binding, where it is only relevant for the binary mode, and removed the AMQP format.

### CODE REVIEW

File: amqp-protocol-binding.md [hunk 1]
```diff
@@ -68,7 +65,7 @@ modes.
 
 In the _structured_ content mode, event metadata attributes and event data are
 placed into the AMQP message's [application data][data] section using an
-[event format](#14-event-formats).
+event format as defined in the CloudEvents [spec][ce].
 
 In the _binary_ content mode, the value of the event `data` is placed
 into the AMQP message's [application data][data] section as-is, with the
```
Comment (alanconway) [inline amqp-protocol-binding.md]:
```
- declaring its media type;
+ declaring its media type mapped to the AMQP `content-type` message property;
```

File: amqp-protocol-binding.md [hunk 2]
```diff
@@ -94,33 +89,14 @@ mandate specific existing features to be used.
 This specification does not further define any of the [CloudEvents][ce] event
 attributes.
 
-Two of the event attributes, `datacontenttype` and `data` are handled specially
-and mapped onto AMQP constructs, all other attributes are transferred as
+One event attribute, `datacontenttype` is handled specially in *binary* content
+mode and mapped onto AMQP constructs. All other attributes are transferred as
```
Comment (alanconway) [inline amqp-protocol-binding.md]: "mapped on to the AMQP content-type message property" - "AMQP constructs" is needlessly vague, it's clearer to be explicit.

File: amqp-protocol-binding.md [hunk 3]
```diff
@@ -151,13 +127,25 @@ directly to the CloudEvents `datacontenttype` attribute.
...
+expectation is that the event data is made available as [UTF-8][rfc3629] encoded
+JSON text for use in AMQP.
+
 #### 3.1.3. Metadata Headers
 
-All [CloudEvents][ce] attributes with exception of `datacontenttype` and `data`
-MUST be individually mapped to and from the AMQP
+All [CloudEvents][ce] attributes with exception of `datacontenttype` MUST be
+individually mapped to and from the AMQP
 [application-properties][app-properties] section, with exceptions noted below.
```
Comment (alanconway) [inline amqp-protocol-binding.md]: Strike ", with exceptions noted below", we already mentioned datacontentype and there aren't any more exceptions.

File: amqp-protocol-binding.md [hunk 4]
```diff
@@ -186,8 +174,32 @@ Examples:
...
+| ------------- | --------------------------- |
+| Boolean       | [boolean][amqp-boolean]     |
+| Integer       | [long][amqp-long]           |
+| String        | [string][amqp-string]       |
+| Binary        | [binary][amqp-binary]       |
+| URI           | [string][amqp-string]       |
+| URI-reference | [string][amqp-string]       |
+| Timestamp     | [timestamp][amqp-timestamp] |
+
+A CloudEvents AMQP *binary* mode implementation MUST allow for attribute values
```
Comment (alanconway) [inline amqp-protocol-binding.md]: That types be "convertible" is a property of an SDK,  not the protocol binding - the core spec already requires it. 
I'd say something like "all attribute values in an AMQP binary message MAY be represented using the native AMQP types above, or MAY be in canonical string form. An implementation MUST be able to interpret both forms on an incoming AMQP message. An implementation SHOULD use the native AMQP form on outgoing AMQP messages when it is efficient to do so, but MAY forward values as canonical strings. A robust implementation MAY further relax the requirements for incoming messages (for example accepting numeric types other than AMQP long) but MUST use only the prescribed AMQP types or canonical strings on outgoing messages.

**Comment by duglin** [comment]
```
./documented-extensions.md: Can't find: ./amqp-transport-binding.md
```
bad link now

File: README.md [hunk 5]
```diff
@@ -31,7 +31,7 @@ The following documents are available:
 |                               |
 | **Optional Specifications:**  |
 | AMQP Event Format             |      [v1.0-rc1](https://github.com/cloudevents/spec/blob/v1.0-rc1/amqp-format.md)       |      [master](https://github.com/cloudevents/spec/blob/master/amqp-format.md)       |
-| AMQP Transport Binding        | [v1.0-rc1](https://github.com/cloudevents/spec/blob/v1.0-rc1/amqp-transport-binding.md) | [master](https://github.com/cloudevents/spec/blob/master/amqp-transport-binding.md) |
+| AMQP Transport Binding        | [v1.0-rc1](https://github.com/cloudevents/spec/blob/v1.0-rc1/amqp-protocol-binding.md) | [master](https://github.com/cloudevents/spec/blob/master/amqp-protocol-binding.md) |
```
Comment (duglin) [inline README.md]: first one needs to remain with the old name since it's already out there

File: amqp-protocol-binding.md (hunk 4 above)
Comment (deissnerk) [reply amqp-protocol-binding.md]: I like the proposed text and updated my proposal accordingly. Some people might argue with some of the MUST/SHOULD/MAY statements. I like how it is leaning towards the AMQP types without enforcing it.

File: README.md (hunk 5 above)
Comment (deissnerk) [reply README.md]: fixed

File: amqp-protocol-binding.md [hunk 6]
```diff
@@ -186,8 +174,29 @@ Examples:
...
+| URI-reference | [string][amqp-string]       |
+| Timestamp     | [timestamp][amqp-timestamp] |
+
+All attribute values in an AMQP binary message MAY be represented using the
+native AMQP types above, or MAY be in canonical string form. An implementation
+MUST be able to interpret both forms on an incoming AMQP message. An
+implementation SHOULD use the native AMQP form on outgoing AMQP messages when it
+is efficient to do so, but MAY forward values as canonical strings. A robust
+implementation MAY further relax the requirements for incoming messages (for
+example accepting numeric types other than AMQP long) but MUST use only the
```
Comment (duglin) [inline amqp-protocol-binding.md]: is the MUST at the end in conflict with the MAY in the first line?  "MUST use only AMQP types" vs "MAY use AMQP types or MAY use strings"

**Comment by clemensv** [comment]
LGTM

**Comment by duglin** [comment]
Conditionally approved on the 10/10 call with a minor wording change that @deissnerk will make today.

While this missed the 2-day deadline the group decided to approve it on the call and give people until end of day tomorrow (friday) to raise any concerns, then when we get 2 LGTMs we'll merge.

**Comment by duglin** [comment]
LGTM
can I get one more?

**Comment by markpeek** [comment]
LGTM

**Comment by duglin** [comment]
thanks @markpeek 
merging

**Pull request merged**